File: leetcode-submissions/3152_Special_Array_II.cpp
class Solution {
public:
   bool isPossible(vector<int>&nums,int start,int end){
        string prev=nums[start]%2==0?"even":"odd";

        for(int i=start+1;i<=end;i++){
            string current=nums[i]%2==0?"even":"odd";
            if(current==prev){
                return false;
            }
            prev=current;
        }
        return true ;

    }
    vector<bool> isArraySpecial(vector<int>& nums, vector<vector<int>>& queries) {
        vector<bool>ans;
        for(auto &ele:queries){
            if(isPossible(nums,ele[0],ele[1])){
                ans.push_back(true);
            }else{
                ans.push_back(false);
            }
        }
        return ans;
    }
};

File: leetcode-submissions/3152_Special_Array_II.ts
function isArraySpecial(nums: number[], queries: number[][]): boolean[] {
    const breaks:number[]=new Array(nums.length).fill(0);
    const prefix:number[]=new Array(nums.length).fill(0);
    const ans:boolean[]=[];
    for(let i:number=0;i<nums.length-1;i++){
breaks[i]=(Math.floor(nums[i]%2)===Math.floor(nums[i+1]%2))?1:0;
    }
    prefix[0]=breaks[0];
    for(let i:number =1;i<breaks.length;i++){
        prefix[i]=prefix[i-1]+breaks[i];
    }
    for(const query of queries){
        const[start,end]=query;
        if(start===end){
ans.push(true);
        }else if((prefix[end]-prefix[start])===0){
ans.push(true);
        }else {
ans.push(false);
        }

    }
    return ans ;
};